lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1312 @@
+    communication = [{
+        'language' : language_codeable_concept,
+        'preferred': True # Assumes that the project language is the patient's preferred language
+    }]
+
+    net_id = normalize_net_id(record.get('netid'))
+    if net_id:
+        patient_id = generate_hash(net_id)
+    else:
+        patient_id = generate_patient_hash(
+            names       = (record['core_participant_first_name'], record['core_participant_last_name']),
+            gender      = gender,
+            birth_date  = record['core_birthdate'],
+            postal_code = record['core_zipcode'])
+
+    if not patient_id:
+        # Some piece of information was missing, so we couldn't generate a
+        # hash.  Fallback to treating this individual as always unique by using
+        # the REDCap record id.
+        patient_id = generate_hash(f"{REDCAP_URL}{record.project.id}/{record['record_id']}")
+
+    LOG.debug(f"Generated individual identifier {patient_id}")
+
+    patient_identifier = create_identifier(f"{INTERNAL_SYSTEM}/individual", patient_id)
+    patient_resource = create_patient_resource([patient_identifier], gender, communication)
+
+    return create_entry_and_reference(patient_resource, "Patient")
+
+
+def create_encounter(record: REDCapRecord, patient_reference: dict,
+    site_reference: dict, locations: list, event_type: EventType,
+    collection_method: CollectionMethod) -> tuple:
+    """
+    Returns a FHIR Encounter resource entry and reference for the encounter in the study.
+    """
+
+    def grab_symptom_key(key: str, variable_name: str) -> Optional[Match[str]]:
+        if record[key] == '1':
+            return re.match(f"{variable_name}___[a-z_]+", key)
+        else:
+            return None
+
+    def build_condition(symptom: str, onset_date: str) -> dict:
+        return create_resource_condition(record, symptom, patient_reference, onset_date)
+
+    def build_diagnosis(symptom: str) -> Optional[dict]:
+        mapped_symptom = map_symptom(symptom)
+        if not mapped_symptom:
+            return None
+
+        return { "condition": { "reference": f"#{mapped_symptom}" } }
+
+    def build_locations_list(location: dict) -> dict:
+        return {
+            "location": create_reference(
+                reference_type = "Location",
+                reference = location["fullUrl"]
+            )
+        }
+
+    def non_tract_locations(resource: dict):
+        return bool(resource) \
+            and resource['resource']['identifier'][0]['system'] != f"{INTERNAL_SYSTEM}/location/tract"
+
+    # Map the various symptoms variables to their onset date.
+    # For daily_symptoms_covid_like use the attestation_date - 24 hours because the question asks
+    # "in the past 24 hours"
+    if event_type == EventType.ENCOUNTER:
+        symptom_onset_map = {
+            'daily_symptoms_covid_like': (datetime.strptime(record['attestation_date'], '%Y-%m-%d') \
+                - timedelta(hours=24)).strftime('%Y-%m-%d'),
```

Comment:
I thought about this and went with ingesting the daily_symptoms_covid_like as questionnaire items even though they do also get captured as symptoms just for ease of use.
@joverlee521 We need an onset date for these symptoms, I believe. Or can we omit one for these?
If we do have to supply an onset date, what do you suggest for these since we don't ask the participant?

Reply:
@joverlee521 Using None works throughout the entire pipeline. The FHIR ETL just doesn't set the onset date. 
So I removed this and am using None for the daily_symptoms_covid_like onset date.

Reply:
I'm no longer ingesting the symptom questions as questionnaire items.

Reply:
@kairstenfay @jover On the daily attestation questionnaire the question is "Which of the following symptoms have you experienced in the last 24 hours? Select all that apply.". There is no question for onset date for these symptoms.
Originally, I was using calculating the onset date as the attestation_date minus 24 hours. Jover questioned that, so I removed it and left no onset date.
Kairsten just pointed out that having no onset date violates FHIR. 
I see two options: Don't use these as symptoms or calculate the onset date.

As Jover points out, there is another very similar question on the same questionnaire:
"Which new or worsening symptoms have you experienced in the last 7 days? Pick any that answer the question."
It has a corresponding onset date field. I am already using that one for symptoms.

Reply:
Thanks for talking about this today. I'm keeping both sets of symptoms. For the daily_covid ones (past 24 hours) I'm simply not setting the onset_date if there isn't one. @kairstenfay thanks for your note. I'm sorry I was grocking what you meant earlier.